pkg/controller/deployment/util/deployment_util.go

```diff
@@ -714,6 +714,17 @@ func GetAvailableReplicaCountForReplicaSets(replicaSets []*apps.ReplicaSet) int3
 	return totalAvailableReplicas
 }
 
+// GetTerminatingReplicaCountForReplicaSets returns the number of terminating pods for all replica sets.
+func GetTerminatingReplicaCountForReplicaSets(replicaSets []*apps.ReplicaSet) int32 {
```

Comment:
```suggestion
func GetTerminatingReplicaCountForReplicaSets(replicaSets []*apps.ReplicaSet) *int32 {
```

Reply:
we have to be able to distinguish between "the number of terminating replicas is unknown" and "there are no terminating replicas"... maybe not a pointer, but then a second return variable or something

Reply:
I think we can add that later if we find it necessary/useful. Which as mentioned in https://github.com/kubernetes/kubernetes/pull/128546#discussion_r1832949128 I think it is not in this PR.

Reply:
this has to return *int32 to distinguish between scenarios where we *can* sum across replicasets and scenarios where we cannot. We can special-case handling of replicasets never synced by the controller at all and treat terminatingReplicas as 0 in that case, see the comment below.

Reply:
Done, returning a pointer and an error now.